sunpy/net/dataretriever/sources/vsm.py

```diff
@@ -0,0 +1,163 @@
+    """
+    Returns a list of URLS to SOLIS VSM files corresponding to value of input timerange.
+    URL source: `http://gong2.nso.edu/pubkeep/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Fixed argument - 'vsm' ,case insensitive.
+
+    Wavelength: Any of 6302, 8542 or 10830 Angstrom units or equivalent values in different units.
+                Expects a Wavelength object.
+                e.g. 6302*u.AA, 630.2*u.nm etc.
+
+    Physobs: Optional, includes 'LOS_MAGNETIC_FIELD', "VECTOR_MAGNETIC_FIELD'
+             'EQUIVALENT_WIDTH'.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> res = Fido.search(a.Time('2016/6/4', '2016/6/4 00:10:00'), a.Instrument('vsm'),
+                          a.Wavelength(6302*u.AA), a.Physobs('VECTOR_MAGNETIC_FIELD'))
+    >>> print (res)
+    [<Table length=6>
+         Start Time           End Time      Source Instrument
+           str19               str19         str5     str3
+    ------------------- ------------------- ------ ----------
+    2015-01-03 00:00:00 2015-01-04 00:00:00  SOLIS        vsm
+    2015-01-04 00:00:00 2015-01-05 00:00:00  SOLIS        vsm
+    2015-01-05 00:00:00 2015-01-06 00:00:00  SOLIS        vsm
+    2015-01-06 00:00:00 2015-01-07 00:00:00  SOLIS        vsm
+    2015-01-07 00:00:00 2015-01-08 00:00:00  SOLIS        vsm
+    2015-01-08 00:00:00 2015-01-09 00:00:00  SOLIS        vsm]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        table_physobs = ['EQUIVALENT_WIDTH', 'LOS_MAGNETIC_FIELD', 'VECTOR_MAGNETIC_FIELD']
+        table_wave = {6302:'v72', 8542:'v82', 10830:'v22'}
+
+        physobs_in = ('physobs' in kwargs.keys() and kwargs['physobs'] in table_physobs)
+##        url_pattern = 'http://gong2.nso.edu/pubkeep/{dtype}/%Y%m/k4{dtype}%y%m%d/k4{dtype}%y%m%dt%H%M%S{suffix}'
+
+        wave_float = kwargs['wavelength'].min.value
+
+        #The isclose function in numpy can only check two arrays if they
+        #are close to each other within some precision. Standalone values such
+        #as int, doubles etc can't be checked that way. In order to do that, we put the
+        #two indiviual values in two seperate arrays da and db and then apply
+        #isclose on both those arrays.
+        da = list()
+        da.append(wave_float)
+        for wave_nums in table_wave.keys():
+            db = list()
+            db.append(wave_nums)
+            if np.isclose(da, db, 1e-10, 1e-10):
+                wave = wave_nums
+                break
+
+        if wave is None:
+            raise ValueError("Enter correct wavelength values and units")
+
+        def download_from_nso(dtype, suffix, timerange):
```

Comment:
I'm not too happy to call `dtype` to this, as it could be confused with `dtype` in the sense of `int`, `float`, etc.
Maybe `obstype` makes more sense.